modules/dop/services/pipeline/pipeline.go

```diff
@@ -634,6 +658,57 @@ func (p *Pipeline) PipelineCronUpdate(req apistructs.GittarPushPayloadEvent) err
 	return nil
 }
 
+func (p *Pipeline) createCron(appDto *apistructs.ApplicationDTO, ymlPathName string, branch string, event apistructs.GittarPushPayloadEvent) error {
+	createV1 := apistructs.PipelineCreateRequest{
+		AppID:             appDto.ID,
+		Branch:            branch,
+		PipelineYmlName:   ymlPathName,
+		Source:            apistructs.PipelineSourceDice,
+		PipelineYmlSource: apistructs.PipelineYmlSourceGittar,
+		UserID:            event.Content.Pusher.Id,
+	}
+	createV2, err := p.ConvertPipelineToV2(&createV1)
+	if err != nil {
+		return fmt.Errorf("ConvertPipelineToV2 error %v req %v", err, createV1)
+	}
+
+	var req = &sourcepb.PipelineSourceListRequest{}
+	req.SourceType = "erda"
+	req.Remote = filepath.Join(appDto.OrgName, appDto.ProjectName, appDto.Name)
+	req.Ref = branch
+	req.Path = func() string {
+		if filepath.Dir(ymlPathName) == "." {
+			return ""
+		}
+		return filepath.Dir(ymlPathName)
+	}()
+	req.Name = strings.Replace(ymlPathName, req.Path+"/", "", 1)
+	result, err := p.pipelineSource.List(context.Background(), req)
```

Comment:
maybe you should use apis.context with dop as internal client.
otherwise you should occur authorization issue.

Reply:
The internal client has been added, and the pipeline level verification will be added later